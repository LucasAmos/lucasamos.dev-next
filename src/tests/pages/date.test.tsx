/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import Date from "../../components/date";
import "@testing-library/jest-dom";

it("renders correctly", () => {
  const { container } = render(<Date dateString="2020-09-19" />);
  expect(container).toMatchSnapshot();
});

describe("Date", () => {
  it("renders correct date", () => {
    const { getByTestId } = render(<Date dateString="2020-09-19" />);

    const date: HTMLElement = getByTestId("date");

    expect(date).toHaveTextContent("September 19, 2020");
  });
});
