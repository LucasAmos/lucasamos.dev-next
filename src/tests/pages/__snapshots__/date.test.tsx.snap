// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <time
    data-testid="date"
    datetime="2020-09-19"
  >
    September 19, 2020
  </time>
</div>
`;
